src/accountsdb/db.zig

```diff
@@ -363,8 +372,16 @@ pub const AccountsDB = struct {
             const file_id_u32: u32 = @intCast(accounts_file_id);
             file_map.putAssumeCapacityNoClobber(file_id_u32, accounts_file);
 
-            if (file_count % 100 == 0 or (file_names.len - file_count) < 100) {
-                printTimeEstimate(&timer, file_names.len, file_count, "reading account files", null);
+            if (print_progress and progress_timer.read() > DB_PROGRESS_UPDATES_NS) {
+                printTimeEstimate(
+                    self.logger,
+                    &timer,
+                    file_names.len,
+                    file_count,
+                    "loading account files",
+                    "thread0",
+                );
+                progress_timer.reset();
```

Comment:
This logic looks good, but it could be written more concisely using `progress_timer.lap()`, because then you won't need to call `progress_timer.reset()` at the end.

Reply:
the problem with `lap()` is if the loop is faster than the threshold, the print code block will never be hit

Reply:
oh oops. it seems so obvious now, what was i thinking